{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c30d9ea0",
   "metadata": {},
   "source": [
    "# Junção CSV's Civic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d24ce13",
   "metadata": {},
   "source": [
    "Infelizmente, os dados que a API oferece não são suficientes para a realização da nossa análise. Por isso, usaremos dados fornecidos pelo CIVIC em formato TSV, que precisam ser convertidos para CSVs para que possamos trabalhar com esses dados. Assim como na API, os arquivos TSV também estão separados e precisam ser mesclados. Para isso, iremos utilizar a função \"merge\", que é usada para combinar dois ou mais conjuntos de dados com base em uma chave comum. Para essa etapa, precisamos apenas da importação do Pandas."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2e690696",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Importação\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f98e504f",
   "metadata": {},
   "source": [
    "A função \"merge\" receberá como entrada dois dataframes, \"df_Assertion\" e \"df_Evidences\", e retornará um novo dataframe que contém os dados combinados. Como mencionado anteriormente, os dados estão em formato TSV, mas foram convertidos para CSV com o uso de uma ferramenta externa. Abaixo, definiremos os arquivos que iremos utilizar."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0aa12809",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Definiçao dos arquivos que serão utilizados\n",
    "df_Variant = pd.read_csv('01-Apr-2022-VariantSummaries.csv')\n",
    "df_Evidences = pd.read_csv('01-Apr-2022-ClinicalEvidenceSummaries.csv')\n",
    "df_Gene = pd.read_csv('01-Apr-2022-GeneSummaries.csv')\n",
    "#df_Assertion = pd.read_csv('01-Apr-2022-AssertionSummaries.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc639be8",
   "metadata": {},
   "source": [
    "Possuímos quatro DataFrames diferentes: df_Variant, que possui **1533** linhas e **33** colunas, df_evidences, que possui **3897** linhas e **42** colunas, df_gene com **477** linhas e **7** colunas e, por último, df_Assertion com **22** linhas e **28** colunas.\n",
    "\n",
    "Antes de podermos realizar o merge, precisamos fazer um pequeno tratamento nos dados fornecidos. Inicialmente, precisamos trocar o nome de duas colunas do df_Assertion, que possuem os mesmos valores e significado no df_evidences, porém com nomes diferentes. Para isso, usaremos o método \"rename\"."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ae7df362",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Trocar o nome das colunas.\n",
    "#df_Assertion.rename(columns={'assertion_type': 'evidence_type'}, inplace = True)\n",
    "#df_Assertion.rename(columns={'assertion_direction': 'evidence_direction'}, inplace = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff6f411e",
   "metadata": {},
   "source": [
    "Após trocarmos os nomes dessas colunas, precisamos apagar algumas outras do mesmo df, pois essas colunas não se encontram em nenhum outro df, e por conta disso, a mesclagem se torna um problema. Por isso, essas colunas estão sendo excluídas, mas nenhum dado importante será perdido."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e47a68a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Apagando colunas\n",
    "# df_Assertion = df_Assertion.drop(['acmg_codes', 'amp_category', 'nccn_guideline', \n",
    "#                                   'fda_companion_test', 'assertion_summary', 'assertion_description',\n",
    "#                                   'assertion_id', 'fda_companion_test', 'regulatory_approval',\n",
    "#                                   'assertion_civic_url', 'evidence_item_ids', 'nccn_guideline_version', 'evidence_items_civic_url' ],axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2fd6d12",
   "metadata": {},
   "source": [
    "Com as colunas apagadas, podemos realizar o primeiro merge. Inicialmente, iremos juntar o df_Assertion tratado e o df_evidences, ambos com a maior quantidade de linhas e colunas. Para essa mesclagem, não é necessário passar uma chave específica, podemos simplesmente usar o método \"how='outer'\". Assim, criaremos um df nomeado Civic que possui a junção de todos os registros dos dois conjuntos de dados originais, mantendo os valores nulos onde não há correspondência entre as chaves de junção, por esse motivo as colunas foram excluídas e tiveram seus nomes trocados.\n",
    "\n",
    "Usaremos esse método, pois desejamos manter todas as informações de ambos os DataFrames. Caso esse não fosse o objetivo, poderíamos ter usado outros parâmetros, como \"inner\", \"left\" ou \"right\"."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a136aea7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Merge entre df_Assertion e df_Evidences\n",
    "#Civic = pd.merge(df_Assertion, df_Evidences, how='outer')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3344e11a",
   "metadata": {},
   "source": [
    "Com a primeira junção finalizada, realizaremos a segunda juntando o novo DataFrame nomeado Civic com o df_Variant. Para que essa mesclagem funcione, usaremos como chave a coluna 'variant_id' e como parâmetro usaremos o left_on e o right_on, fazendo o uso desses parâmetros teremos especificado as colunas que serão utilizadas como chaves de junção para o conjunto de dados.\n",
    "\n",
    "Para que nossos dados não fiquem poluídos e que não precisemos ter um trabalho extra excluindo colunas repetidas que serão geradas com o merge, pois nosso conjunto de dados possuem colunas com o mesmo nome, iremos definir o parâmetro \"suffixes\" como uma tupla vazia e uma tupla com o sufixo \"_y\", o que garantirá que não haverá colunas duplicadas após a junção.\n",
    "\n",
    "Após a junção, usaremos o método \"filter\" para selecionar apenas as colunas que possuem o sufixo \"_y\" no nome e usaremos o método \"drop\" para removê-las."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "788e19d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_Variant.rename(columns={'summary': 'variant_summary'}, inplace = True)\n",
    "df_Variant = df_Variant.drop(['assertion_ids', 'is_flagged'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3864a66d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>drugs</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>...</th>\n",
       "      <th>representative_transcript2</th>\n",
       "      <th>ensembl_version</th>\n",
       "      <th>reference_build</th>\n",
       "      <th>variant_summary</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>evidence_civic_url</th>\n",
       "      <th>variant_civic_url</th>\n",
       "      <th>gene_civic_url</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2020-11-20 16:18:21 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/64</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/28</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>5156</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>PDGFRA D842 mutations are characterized broadl...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2015-06-21 16:49:38 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/99</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/38</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>R882</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>DNMT3A R882 mutations are associated with cyto...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2016-03-26 22:10:59 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/32</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/18</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>R882</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>DNMT3A R882 mutations are associated with cyto...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2016-03-16 22:09:27 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/32</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/18</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2015-06-21 16:49:39 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/64</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/28</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3892</th>\n",
       "      <td>ERCC1</td>\n",
       "      <td>2067</td>\n",
       "      <td>Underexpression</td>\n",
       "      <td>Ovary Epithelial Cancer</td>\n",
       "      <td>2152.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Platinum Compound</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2022-03-01 18:44:57 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/1735</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3893</th>\n",
       "      <td>NTRK1</td>\n",
       "      <td>4914</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Solid Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Larotrectinib</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2022-02-24 18:44:37 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/3983</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3894</th>\n",
       "      <td>CTCF</td>\n",
       "      <td>10664</td>\n",
       "      <td>K365T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Functional</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2022-03-08 18:33:47 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/8508</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3895</th>\n",
       "      <td>KIT</td>\n",
       "      <td>3815</td>\n",
       "      <td>EXPRESSION</td>\n",
       "      <td>Glioblastoma</td>\n",
       "      <td>3068.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sunitinib</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2022-03-15 18:35:48 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/429</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/29</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3896</th>\n",
       "      <td>SSTR5</td>\n",
       "      <td>6755</td>\n",
       "      <td>expression</td>\n",
       "      <td>Neuroendocrine Tumor</td>\n",
       "      <td>169.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pasireotide</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2022-03-15 18:09:18 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/5499</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3897 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene  entrez_id          variant                         disease  \\\n",
       "0       JAK2       3717            V617F               Lymphoid Leukemia   \n",
       "1     PDGFRA       5156            D842V  Gastrointestinal Stromal Tumor   \n",
       "2     DNMT3A       1788             R882          Acute Myeloid Leukemia   \n",
       "3     DNMT3A       1788             R882          Acute Myeloid Leukemia   \n",
       "4       JAK2       3717            V617F        Chronic Myeloid Leukemia   \n",
       "...      ...        ...              ...                             ...   \n",
       "3892   ERCC1       2067  Underexpression         Ovary Epithelial Cancer   \n",
       "3893   NTRK1       4914    Amplification                     Solid Tumor   \n",
       "3894    CTCF      10664            K365T                             NaN   \n",
       "3895     KIT       3815       EXPRESSION                    Glioblastoma   \n",
       "3896   SSTR5       6755       expression            Neuroendocrine Tumor   \n",
       "\n",
       "        doid phenotypes              drugs drug_interaction_type  \\\n",
       "0     1037.0        NaN                NaN                   NaN   \n",
       "1     9253.0        NaN                NaN                   NaN   \n",
       "2     9119.0        NaN                NaN                   NaN   \n",
       "3     9119.0        NaN                NaN                   NaN   \n",
       "4     8552.0        NaN                NaN                   NaN   \n",
       "...      ...        ...                ...                   ...   \n",
       "3892  2152.0        NaN  Platinum Compound                   NaN   \n",
       "3893     NaN        NaN      Larotrectinib                   NaN   \n",
       "3894     NaN        NaN                NaN                   NaN   \n",
       "3895  3068.0        NaN          Sunitinib                   NaN   \n",
       "3896   169.0        NaN        Pasireotide                   NaN   \n",
       "\n",
       "     evidence_type evidence_direction  ... representative_transcript2  \\\n",
       "0       Diagnostic           Supports  ...                        NaN   \n",
       "1       Diagnostic           Supports  ...                        NaN   \n",
       "2       Diagnostic           Supports  ...                        NaN   \n",
       "3       Diagnostic           Supports  ...                        NaN   \n",
       "4       Diagnostic           Supports  ...                        NaN   \n",
       "...            ...                ...  ...                        ...   \n",
       "3892    Predictive           Supports  ...                        NaN   \n",
       "3893    Predictive           Supports  ...                        NaN   \n",
       "3894    Functional           Supports  ...                        NaN   \n",
       "3895    Predictive           Supports  ...                        NaN   \n",
       "3896    Predictive           Supports  ...                        NaN   \n",
       "\n",
       "     ensembl_version reference_build  \\\n",
       "0               75.0          GRCh37   \n",
       "1               75.0          GRCh37   \n",
       "2               75.0          GRCh37   \n",
       "3               75.0          GRCh37   \n",
       "4               75.0          GRCh37   \n",
       "...              ...             ...   \n",
       "3892            75.0          GRCh37   \n",
       "3893            75.0          GRCh37   \n",
       "3894             NaN             NaN   \n",
       "3895            75.0          GRCh37   \n",
       "3896             NaN             NaN   \n",
       "\n",
       "                                        variant_summary variant_origin  \\\n",
       "0     JAK2 V617F is a highly recurrent mutation in m...        Somatic   \n",
       "1     PDGFRA D842 mutations are characterized broadl...        Somatic   \n",
       "2     DNMT3A R882 mutations are associated with cyto...        Somatic   \n",
       "3     DNMT3A R882 mutations are associated with cyto...        Somatic   \n",
       "4     JAK2 V617F is a highly recurrent mutation in m...        Somatic   \n",
       "...                                                 ...            ...   \n",
       "3892                                                NaN            NaN   \n",
       "3893                                                NaN        Somatic   \n",
       "3894                                                NaN            NaN   \n",
       "3895                                                NaN        Somatic   \n",
       "3896                                                NaN        Somatic   \n",
       "\n",
       "             last_review_date  \\\n",
       "0     2020-11-20 16:18:21 UTC   \n",
       "1     2015-06-21 16:49:38 UTC   \n",
       "2     2016-03-26 22:10:59 UTC   \n",
       "3     2016-03-16 22:09:27 UTC   \n",
       "4     2015-06-21 16:49:39 UTC   \n",
       "...                       ...   \n",
       "3892  2022-03-01 18:44:57 UTC   \n",
       "3893  2022-02-24 18:44:37 UTC   \n",
       "3894  2022-03-08 18:33:47 UTC   \n",
       "3895  2022-03-15 18:35:48 UTC   \n",
       "3896  2022-03-15 18:09:18 UTC   \n",
       "\n",
       "                                     evidence_civic_url  \\\n",
       "0     https://civic.genome.wustl.edu/links/evidence_...   \n",
       "1     https://civic.genome.wustl.edu/links/evidence_...   \n",
       "2     https://civic.genome.wustl.edu/links/evidence_...   \n",
       "3     https://civic.genome.wustl.edu/links/evidence_...   \n",
       "4     https://civic.genome.wustl.edu/links/evidence_...   \n",
       "...                                                 ...   \n",
       "3892  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "3893  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "3894  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "3895  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "3896  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "\n",
       "                                      variant_civic_url  \\\n",
       "0      https://civic.genome.wustl.edu/links/variants/64   \n",
       "1      https://civic.genome.wustl.edu/links/variants/99   \n",
       "2      https://civic.genome.wustl.edu/links/variants/32   \n",
       "3      https://civic.genome.wustl.edu/links/variants/32   \n",
       "4      https://civic.genome.wustl.edu/links/variants/64   \n",
       "...                                                 ...   \n",
       "3892  https://civic.genome.wustl.edu/links/variants/...   \n",
       "3893  https://civic.genome.wustl.edu/links/variants/...   \n",
       "3894  https://civic.genome.wustl.edu/links/variants/...   \n",
       "3895  https://civic.genome.wustl.edu/links/variants/429   \n",
       "3896  https://civic.genome.wustl.edu/links/variants/...   \n",
       "\n",
       "                                       gene_civic_url is_flagged  \n",
       "0       https://civic.genome.wustl.edu/links/genes/28      False  \n",
       "1       https://civic.genome.wustl.edu/links/genes/38      False  \n",
       "2       https://civic.genome.wustl.edu/links/genes/18      False  \n",
       "3       https://civic.genome.wustl.edu/links/genes/18      False  \n",
       "4       https://civic.genome.wustl.edu/links/genes/28      False  \n",
       "...                                               ...        ...  \n",
       "3892  https://civic.genome.wustl.edu/links/genes/1735      False  \n",
       "3893  https://civic.genome.wustl.edu/links/genes/3983      False  \n",
       "3894  https://civic.genome.wustl.edu/links/genes/8508      False  \n",
       "3895    https://civic.genome.wustl.edu/links/genes/29      False  \n",
       "3896  https://civic.genome.wustl.edu/links/genes/5499      False  \n",
       "\n",
       "[3897 rows x 42 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_Evidences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1791824a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene', 'entrez_id', 'variant', 'disease', 'doid', 'phenotypes',\n",
       "       'drugs', 'drug_interaction_type', 'evidence_type', 'evidence_direction',\n",
       "       'evidence_level', 'clinical_significance', 'evidence_statement',\n",
       "       'citation_id', 'source_type', 'asco_abstract_id', 'citation', 'nct_ids',\n",
       "       'rating', 'evidence_status', 'evidence_id', 'variant_id', 'gene_id',\n",
       "       'chromosome', 'start', 'stop', 'reference_bases', 'variant_bases',\n",
       "       'representative_transcript', 'chromosome2', 'start2', 'stop2',\n",
       "       'representative_transcript2', 'ensembl_version', 'reference_build',\n",
       "       'variant_summary', 'variant_origin', 'last_review_date',\n",
       "       'evidence_civic_url', 'variant_civic_url', 'gene_civic_url',\n",
       "       'is_flagged'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_Evidences.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "58070585",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['variant_id', 'variant_civic_url', 'gene', 'entrez_id', 'variant',\n",
       "       'variant_summary', 'variant_groups', 'chromosome', 'start', 'stop',\n",
       "       'reference_bases', 'variant_bases', 'representative_transcript',\n",
       "       'ensembl_version', 'reference_build', 'chromosome2', 'start2', 'stop2',\n",
       "       'representative_transcript2', 'variant_types', 'hgvs_expressions',\n",
       "       'last_review_date', 'civic_variant_evidence_score',\n",
       "       'allele_registry_id', 'clinvar_ids', 'variant_aliases',\n",
       "       'assertion_civic_urls', 'Unnamed: 29', 'Unnamed: 30', 'Unnamed: 31',\n",
       "       'Unnamed: 32'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_Variant.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "acf9eb99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene_id', 'gene', 'entrez_id', 'description'], dtype='object')"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_Gene.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "81f80d8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Merge entre Civic e df_Variant\n",
    "Civic = pd.merge(df_Evidences, df_Variant,  left_on='variant_id', right_on='variant_id', how='left', suffixes=('','_y'))\n",
    "Civic.drop(Civic.filter(regex='_y$').columns, axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1eb0ee89",
   "metadata": {},
   "source": [
    "Com três DataFrames iniciais juntos, seguiremos para o df_genes. Assim como o df_Assertion, ele precisa ser tratado antes da junção, porém por motivos diferentes. O df_genes possui valores duplicados que se encontram em outros DataFrames. Esses valores serão excluídos, e manteremos apenas as colunas \"description\", que não possuímos em outros DataFrames, e \"name\", que usaremos como chave de junção."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "03e51edc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Apagando colunas\n",
    "df_Gene = df_Gene.drop(['gene_civic_url', 'last_review_date', 'is_flagged'],axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0887cdb2",
   "metadata": {},
   "source": [
    "Também não possuímos a coluna \"name\" em nenhum outro dataframe, mas seus valores correspondem às colunas denominadas como \"gene\" em outros dataframes, por isso faremos um rename nesse dataframe.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ba18ab8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Trocando o nome da coluna name para gene.\n",
    "df_Gene.rename(columns={'name': 'gene'}, inplace = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ea8c8fc",
   "metadata": {},
   "source": [
    "Com o \"df_gene\" tratado, realizaremos o último merge juntando o conjunto total ao último dataframe. Usaremos as colunas \"gene\" como chave de junção e passaremos o mesmo parâmetro \"suffixes\" para apagar colunas duplicadas."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c468f492",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Merge entre Civic e df_gene\n",
    "Civic = pd.merge(Civic, df_Gene,  left_on='gene', right_on='gene', how='left', suffixes=('','_y'))\n",
    "Civic.drop(Civic.filter(regex='_y$').columns, axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03947854",
   "metadata": {},
   "source": [
    "Agora, com o dataframe pronto, realizaremos o último tratamento dessa etapa. Aqui, apagaremos colunas sem valor para nossa análise, como URLs e a última edição (\"last_review_date\"), colunas com valores nulos em todos os dataframes e colunas que não serão usadas."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "6e4e915f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>drugs</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>...</th>\n",
       "      <th>variant_summary</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>is_flagged</th>\n",
       "      <th>variant_types</th>\n",
       "      <th>hgvs_expressions</th>\n",
       "      <th>civic_variant_evidence_score</th>\n",
       "      <th>allele_registry_id</th>\n",
       "      <th>clinvar_ids</th>\n",
       "      <th>variant_aliases</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>False</td>\n",
       "      <td>missense_variant,gain_of_function_variant</td>\n",
       "      <td>NC_000009.11:g.5073770G&gt;T,NM_004972.3:c.1849G&gt;...</td>\n",
       "      <td>83.0</td>\n",
       "      <td>CA124183</td>\n",
       "      <td>14662</td>\n",
       "      <td>VAL617PHE,RS77375493</td>\n",
       "      <td>JAK2 is a kinase that is misregulated or mutat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>5156</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>PDGFRA D842 mutations are characterized broadl...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>False</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006206.4:c.2525A&gt;T,NP_006197.1:p.Asp842Val,...</td>\n",
       "      <td>100.5</td>\n",
       "      <td>CA123194</td>\n",
       "      <td>13543</td>\n",
       "      <td>ASP842VAL,RS121908585</td>\n",
       "      <td>Commonly mutated in GI tract tumors, PDGFR fam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>R882</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>DNMT3A R882 mutations are associated with cyto...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>False</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>510.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>375882,375883,375884,375879,375880,375881</td>\n",
       "      <td>R693,ARG882,ARG693</td>\n",
       "      <td>DNMT3A is one of several epigenetic modifiers ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>R882</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>DNMT3A R882 mutations are associated with cyto...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>False</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>510.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>375882,375883,375884,375879,375880,375881</td>\n",
       "      <td>R693,ARG882,ARG693</td>\n",
       "      <td>DNMT3A is one of several epigenetic modifiers ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>False</td>\n",
       "      <td>missense_variant,gain_of_function_variant</td>\n",
       "      <td>NC_000009.11:g.5073770G&gt;T,NM_004972.3:c.1849G&gt;...</td>\n",
       "      <td>83.0</td>\n",
       "      <td>CA124183</td>\n",
       "      <td>14662</td>\n",
       "      <td>VAL617PHE,RS77375493</td>\n",
       "      <td>JAK2 is a kinase that is misregulated or mutat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3892</th>\n",
       "      <td>ERCC1</td>\n",
       "      <td>2067</td>\n",
       "      <td>Underexpression</td>\n",
       "      <td>Ovary Epithelial Cancer</td>\n",
       "      <td>2152.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Platinum Compound</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3893</th>\n",
       "      <td>NTRK1</td>\n",
       "      <td>4914</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Solid Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Larotrectinib</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>False</td>\n",
       "      <td>transcript_amplification</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3894</th>\n",
       "      <td>CTCF</td>\n",
       "      <td>10664</td>\n",
       "      <td>K365T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Functional</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3895</th>\n",
       "      <td>KIT</td>\n",
       "      <td>3815</td>\n",
       "      <td>EXPRESSION</td>\n",
       "      <td>Glioblastoma</td>\n",
       "      <td>3068.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sunitinib</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c-KIT activation has been shown to have oncoge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3896</th>\n",
       "      <td>SSTR5</td>\n",
       "      <td>6755</td>\n",
       "      <td>expression</td>\n",
       "      <td>Neuroendocrine Tumor</td>\n",
       "      <td>169.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pasireotide</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3897 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene  entrez_id          variant                         disease  \\\n",
       "0       JAK2       3717            V617F               Lymphoid Leukemia   \n",
       "1     PDGFRA       5156            D842V  Gastrointestinal Stromal Tumor   \n",
       "2     DNMT3A       1788             R882          Acute Myeloid Leukemia   \n",
       "3     DNMT3A       1788             R882          Acute Myeloid Leukemia   \n",
       "4       JAK2       3717            V617F        Chronic Myeloid Leukemia   \n",
       "...      ...        ...              ...                             ...   \n",
       "3892   ERCC1       2067  Underexpression         Ovary Epithelial Cancer   \n",
       "3893   NTRK1       4914    Amplification                     Solid Tumor   \n",
       "3894    CTCF      10664            K365T                             NaN   \n",
       "3895     KIT       3815       EXPRESSION                    Glioblastoma   \n",
       "3896   SSTR5       6755       expression            Neuroendocrine Tumor   \n",
       "\n",
       "        doid phenotypes              drugs drug_interaction_type  \\\n",
       "0     1037.0        NaN                NaN                   NaN   \n",
       "1     9253.0        NaN                NaN                   NaN   \n",
       "2     9119.0        NaN                NaN                   NaN   \n",
       "3     9119.0        NaN                NaN                   NaN   \n",
       "4     8552.0        NaN                NaN                   NaN   \n",
       "...      ...        ...                ...                   ...   \n",
       "3892  2152.0        NaN  Platinum Compound                   NaN   \n",
       "3893     NaN        NaN      Larotrectinib                   NaN   \n",
       "3894     NaN        NaN                NaN                   NaN   \n",
       "3895  3068.0        NaN          Sunitinib                   NaN   \n",
       "3896   169.0        NaN        Pasireotide                   NaN   \n",
       "\n",
       "     evidence_type evidence_direction  ...  \\\n",
       "0       Diagnostic           Supports  ...   \n",
       "1       Diagnostic           Supports  ...   \n",
       "2       Diagnostic           Supports  ...   \n",
       "3       Diagnostic           Supports  ...   \n",
       "4       Diagnostic           Supports  ...   \n",
       "...            ...                ...  ...   \n",
       "3892    Predictive           Supports  ...   \n",
       "3893    Predictive           Supports  ...   \n",
       "3894    Functional           Supports  ...   \n",
       "3895    Predictive           Supports  ...   \n",
       "3896    Predictive           Supports  ...   \n",
       "\n",
       "                                        variant_summary variant_origin  \\\n",
       "0     JAK2 V617F is a highly recurrent mutation in m...        Somatic   \n",
       "1     PDGFRA D842 mutations are characterized broadl...        Somatic   \n",
       "2     DNMT3A R882 mutations are associated with cyto...        Somatic   \n",
       "3     DNMT3A R882 mutations are associated with cyto...        Somatic   \n",
       "4     JAK2 V617F is a highly recurrent mutation in m...        Somatic   \n",
       "...                                                 ...            ...   \n",
       "3892                                                NaN            NaN   \n",
       "3893                                                NaN        Somatic   \n",
       "3894                                                NaN            NaN   \n",
       "3895                                                NaN        Somatic   \n",
       "3896                                                NaN        Somatic   \n",
       "\n",
       "     is_flagged                              variant_types  \\\n",
       "0         False  missense_variant,gain_of_function_variant   \n",
       "1         False                           missense_variant   \n",
       "2         False                           missense_variant   \n",
       "3         False                           missense_variant   \n",
       "4         False  missense_variant,gain_of_function_variant   \n",
       "...         ...                                        ...   \n",
       "3892      False                                        NaN   \n",
       "3893      False                   transcript_amplification   \n",
       "3894      False                                        NaN   \n",
       "3895      False                                        NaN   \n",
       "3896      False                                        NaN   \n",
       "\n",
       "                                       hgvs_expressions  \\\n",
       "0     NC_000009.11:g.5073770G>T,NM_004972.3:c.1849G>...   \n",
       "1     NM_006206.4:c.2525A>T,NP_006197.1:p.Asp842Val,...   \n",
       "2                                                   NaN   \n",
       "3                                                   NaN   \n",
       "4     NC_000009.11:g.5073770G>T,NM_004972.3:c.1849G>...   \n",
       "...                                                 ...   \n",
       "3892                                                NaN   \n",
       "3893                                                NaN   \n",
       "3894                                                NaN   \n",
       "3895                                                NaN   \n",
       "3896                                                NaN   \n",
       "\n",
       "     civic_variant_evidence_score  allele_registry_id  \\\n",
       "0                            83.0            CA124183   \n",
       "1                           100.5            CA123194   \n",
       "2                           510.0                 NaN   \n",
       "3                           510.0                 NaN   \n",
       "4                            83.0            CA124183   \n",
       "...                           ...                 ...   \n",
       "3892                         15.0                 NaN   \n",
       "3893                          5.0                 NaN   \n",
       "3894                          0.0                 NaN   \n",
       "3895                         20.0                 NaN   \n",
       "3896                          0.0                 NaN   \n",
       "\n",
       "                                    clinvar_ids        variant_aliases  \\\n",
       "0                                         14662   VAL617PHE,RS77375493   \n",
       "1                                         13543  ASP842VAL,RS121908585   \n",
       "2     375882,375883,375884,375879,375880,375881     R693,ARG882,ARG693   \n",
       "3     375882,375883,375884,375879,375880,375881     R693,ARG882,ARG693   \n",
       "4                                         14662   VAL617PHE,RS77375493   \n",
       "...                                         ...                    ...   \n",
       "3892                                        NaN                    NaN   \n",
       "3893                                        NaN                    NaN   \n",
       "3894                                        NaN                    NaN   \n",
       "3895                                        NaN                    NaN   \n",
       "3896                                        NaN                    NaN   \n",
       "\n",
       "                                            description  \n",
       "0     JAK2 is a kinase that is misregulated or mutat...  \n",
       "1     Commonly mutated in GI tract tumors, PDGFR fam...  \n",
       "2     DNMT3A is one of several epigenetic modifiers ...  \n",
       "3     DNMT3A is one of several epigenetic modifiers ...  \n",
       "4     JAK2 is a kinase that is misregulated or mutat...  \n",
       "...                                                 ...  \n",
       "3892                                                NaN  \n",
       "3893                                                NaN  \n",
       "3894                                                NaN  \n",
       "3895  c-KIT activation has been shown to have oncoge...  \n",
       "3896                                                NaN  \n",
       "\n",
       "[3897 rows x 39 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Removendo colunas\n",
    "Civic = Civic.drop(['gene_civic_url', 'Unnamed: 32', 'Unnamed: 31', 'Unnamed: 29',\n",
    "                    'Unnamed: 30', 'assertion_civic_urls', 'variant_civic_url', \n",
    "                    'evidence_civic_url', 'last_review_date', 'asco_abstract_id',\n",
    "                    'nct_ids', 'chromosome2', 'start2', 'stop2', 'representative_transcript2',\n",
    "                    'variant_groups'\n",
    "],axis=1)\n",
    "Civic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d3ddf31",
   "metadata": {},
   "source": [
    "Para que o dataframe \"Civic\" final possa ser utilizado pelo restante do grupo, iremos salvá-lo como CSV."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "8256eb1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Salvando o df Civic como um csv\n",
    "#Civic.to_csv('Civic.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cef4c667",
   "metadata": {},
   "source": [
    "Diante disso, podemos ver que nosso código consiste na leitura de quatro arquivos csv (df_Variant, df_Evidence, df_Gene, Df_Assertion) fornecidos pelo proprio Civic, e na sua mesclagem em um dataframe final chamado Civic, tambem possuimos alguns tratamentos sendo realizados nos dados, como a exclusão de colunas não relevantes, a renomeação de colunas e a remoçao de valores duplicados.\n",
    "Ao final o Dataframe Civic é salvo em um arquivo csv para ser utilizado posteriormente."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
