{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import allel\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.max_rows', 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes        = pd.read_csv('01-Apr-2022-GeneSummaries.csv')\n",
    "df_variants     = pd.read_csv('01-Apr-2022-VariantSummaries.csv')\n",
    "df_evidence     = pd.read_csv('01-Apr-2022-ClinicalEvidenceSummaries.csv')\n",
    "df_assertions   = pd.read_csv('01-Apr-2022-AssertionSummaries.csv')\n",
    "df_variantGroup = pd.read_csv('01-Apr-2022-VariantGroupSummaries.csv')\n",
    "accepted_variants = allel.vcf_to_dataframe('01-Apr-2022-civic_accepted.vcf')\n",
    "accepted_sub = allel.vcf_to_dataframe('01-Apr-2022-civic_accepted_and_submitted.vcf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_civic_url</th>\n",
       "      <th>name</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>description</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/1</td>\n",
       "      <td>ALK</td>\n",
       "      <td>238</td>\n",
       "      <td>ALK amplifications, fusions and mutations have...</td>\n",
       "      <td>2017-03-06 00:00:15 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/2</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>207</td>\n",
       "      <td>AKT1, also referred to as protein kinase B, is...</td>\n",
       "      <td>2017-02-09 21:58:06 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/3</td>\n",
       "      <td>ARAF</td>\n",
       "      <td>369</td>\n",
       "      <td>ARAF has recently become increasingly consider...</td>\n",
       "      <td>2015-06-21 16:49:19 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/4</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>ABL1 is most relevant to cancer in its role in...</td>\n",
       "      <td>2015-06-21 16:49:20 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/5</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>673</td>\n",
       "      <td>BRAF mutations are found to be recurrent in ma...</td>\n",
       "      <td>2019-09-19 19:32:44 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene_id                                gene_civic_url  name  entrez_id  \\\n",
       "0        1  https://civic.genome.wustl.edu/links/genes/1   ALK        238   \n",
       "1        2  https://civic.genome.wustl.edu/links/genes/2  AKT1        207   \n",
       "2        3  https://civic.genome.wustl.edu/links/genes/3  ARAF        369   \n",
       "3        4  https://civic.genome.wustl.edu/links/genes/4  ABL1         25   \n",
       "4        5  https://civic.genome.wustl.edu/links/genes/5  BRAF        673   \n",
       "\n",
       "                                         description         last_review_date  \\\n",
       "0  ALK amplifications, fusions and mutations have...  2017-03-06 00:00:15 UTC   \n",
       "1  AKT1, also referred to as protein kinase B, is...  2017-02-09 21:58:06 UTC   \n",
       "2  ARAF has recently become increasingly consider...  2015-06-21 16:49:19 UTC   \n",
       "3  ABL1 is most relevant to cancer in its role in...  2015-06-21 16:49:20 UTC   \n",
       "4  BRAF mutations are found to be recurrent in ma...  2019-09-19 19:32:44 UTC   \n",
       "\n",
       "   is_flagged  \n",
       "0       False  \n",
       "1       False  \n",
       "2       False  \n",
       "3       False  \n",
       "4       False  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### df_genes\n",
    " - gene_id = parece index mas de fato é o codigo do gene\n",
    " - gene_civic_url = url (está indo para a pagina onde possui todas as informaçoes do gene, incluindo sua descrição)\n",
    " - name = não possui nulo ou duplicado, aparentemente é o nome do gene (na descrição vários são citados)\n",
    " - entrez_id = não possui nulo ou duplicado, entender melhor que id é esse\n",
    " - description = apenas alguns possuem descrição, é a descrição do gene e suas características.(importante para interpretar as predições)\n",
    " - last_review_date = acredito ser irrelevante para o treinamente do modelo.\n",
    " - is_flagged = todos estão como false, facil detecção."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_civic_url</th>\n",
       "      <th>gene</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant</th>\n",
       "      <th>summary</th>\n",
       "      <th>variant_groups</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "      <th>ensembl_version</th>\n",
       "      <th>reference_build</th>\n",
       "      <th>chromosome2</th>\n",
       "      <th>start2</th>\n",
       "      <th>stop2</th>\n",
       "      <th>representative_transcript2</th>\n",
       "      <th>variant_types</th>\n",
       "      <th>hgvs_expressions</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>civic_variant_evidence_score</th>\n",
       "      <th>allele_registry_id</th>\n",
       "      <th>clinvar_ids</th>\n",
       "      <th>variant_aliases</th>\n",
       "      <th>assertion_ids</th>\n",
       "      <th>assertion_civic_urls</th>\n",
       "      <th>is_flagged</th>\n",
       "      <th>Unnamed: 29</th>\n",
       "      <th>Unnamed: 30</th>\n",
       "      <th>Unnamed: 31</th>\n",
       "      <th>Unnamed: 32</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/1</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>BCR::ABL</td>\n",
       "      <td>The BCR-ABL fusion protein, commonly referred ...</td>\n",
       "      <td>ABL1 fusions in B-ALL</td>\n",
       "      <td>22</td>\n",
       "      <td>23522397.0</td>\n",
       "      <td>23632600.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000305877.8</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>9</td>\n",
       "      <td>133729451.0</td>\n",
       "      <td>133763063.0</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "      <td>transcript_fusion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2022-03-10 22:27:25 UTC</td>\n",
       "      <td>255.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T(9;22)(Q34;Q11),BCR-ABL1,BCR-ABL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/2</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>BCR::ABL T315I</td>\n",
       "      <td>While the efficacy of imatinib has revolutioni...</td>\n",
       "      <td>Imatinib Resistance</td>\n",
       "      <td>9</td>\n",
       "      <td>133748283.0</td>\n",
       "      <td>133748283.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant,transcript_fusion</td>\n",
       "      <td>NM_007313.2:c.1001C&gt;T,NP_005148.2:p.Thr315Ile,...</td>\n",
       "      <td>2022-03-10 19:55:19 UTC</td>\n",
       "      <td>146.0</td>\n",
       "      <td>CA122575</td>\n",
       "      <td>12624</td>\n",
       "      <td>THR334ILE,RS121913459,BCR-ABL THR315ILE,BCR-AB...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   variant_id                                variant_civic_url  gene  \\\n",
       "0           1  https://civic.genome.wustl.edu/links/variants/1  ABL1   \n",
       "1           2  https://civic.genome.wustl.edu/links/variants/2  ABL1   \n",
       "\n",
       "   entrez_id         variant  \\\n",
       "0         25        BCR::ABL   \n",
       "1         25  BCR::ABL T315I   \n",
       "\n",
       "                                             summary         variant_groups  \\\n",
       "0  The BCR-ABL fusion protein, commonly referred ...  ABL1 fusions in B-ALL   \n",
       "1  While the efficacy of imatinib has revolutioni...    Imatinib Resistance   \n",
       "\n",
       "  chromosome        start         stop reference_bases variant_bases  \\\n",
       "0         22   23522397.0   23632600.0             NaN           NaN   \n",
       "1          9  133748283.0  133748283.0               C             T   \n",
       "\n",
       "  representative_transcript  ensembl_version reference_build chromosome2  \\\n",
       "0         ENST00000305877.8             75.0          GRCh37           9   \n",
       "1         ENST00000318560.5             75.0          GRCh37         NaN   \n",
       "\n",
       "        start2        stop2 representative_transcript2  \\\n",
       "0  133729451.0  133763063.0          ENST00000318560.5   \n",
       "1          NaN          NaN                        NaN   \n",
       "\n",
       "                        variant_types  \\\n",
       "0                   transcript_fusion   \n",
       "1  missense_variant,transcript_fusion   \n",
       "\n",
       "                                    hgvs_expressions         last_review_date  \\\n",
       "0                                                NaN  2022-03-10 22:27:25 UTC   \n",
       "1  NM_007313.2:c.1001C>T,NP_005148.2:p.Thr315Ile,...  2022-03-10 19:55:19 UTC   \n",
       "\n",
       "   civic_variant_evidence_score allele_registry_id clinvar_ids  \\\n",
       "0                         255.5                NaN         NaN   \n",
       "1                         146.0           CA122575       12624   \n",
       "\n",
       "                                     variant_aliases assertion_ids  \\\n",
       "0                  T(9;22)(Q34;Q11),BCR-ABL1,BCR-ABL           NaN   \n",
       "1  THR334ILE,RS121913459,BCR-ABL THR315ILE,BCR-AB...           NaN   \n",
       "\n",
       "  assertion_civic_urls is_flagged Unnamed: 29 Unnamed: 30 Unnamed: 31  \\\n",
       "0                  NaN      FALSE         NaN         NaN         NaN   \n",
       "1                  NaN      FALSE         NaN         NaN         NaN   \n",
       "\n",
       "  Unnamed: 32  \n",
       "0         NaN  \n",
       "1         NaN  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variants.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### df_variants\n",
    " - variant_id = index\n",
    " - variant_civic_url = o mesmo que ocorre com o genes, detalhes da variante\n",
    " - gene = aqui os genes se repetem, pois mais de uma variante pode pertencer ao mesmo gene.\n",
    " - entrez_id = possui essa informação na tabela gene, possivel chave de ligação.\n",
    " - variant = é a variante do gene, assim como o gene pode possuir varias variantes o contrário também acontece.\n",
    " - summary = descrição e associação da variante em resposta ao câncer (importante para interpretar as predições)\n",
    " - variant_groups = preciso entender o df que consta no outro df\n",
    " - chromosome = poucos valores unicos e alguns nulos (em pesquisa me parece que cada cromossomo tem sua função, necessário entender melhor sobre)\n",
    " - start / stop\n",
    "  \n",
    "  \n",
    "quando entro em detalhes para descobrir o start e stop, sou direcionada para esta pagina\n",
    "está relacionado com a coluna anterior que é o cromossomo, no link de detalhe aponta como cordenada.\n",
    "http://www.ensembl.org/Homo_sapiens/Transcript/Summary?g=ENSG00000186716;r=22:23180509-23318037;t=ENST00000305877\n",
    "\n",
    " - reference_bases = DNA, Adenina, Timina, Citosina e Guanina (está junto as coordenadas do cromossomo)\n",
    " - variant_bases = também sobre sequenciamento de DNA (também está posicionado em coordenadas)\n",
    " - representative_transcript = é o número da trasncrição das coordenadas (exemplo de link no star e stop)\n",
    " - ensembl_version = temos apenas 3 valores e 2 deles não se repetem, não sei ao certo o que significa, se é a versão do teste talvez, ou da transcrição...?\n",
    " - reference_build = está relacionada a coluna anterior, possui apenas 2 valores.\n",
    " - colunas irão se repetir, com valores diferentes, acredito que está associado a um novo teste de variante (colunas: chromosome2, start2, stop2 e representative_transcript2) os nulos aumentam nessa segunda, talvez menos genes foram testados mais de uma vez?\n",
    " - variant_types = nem todas as colunas possuem valores e as que possuem se repete, o gene/variante que possui pode receber varias informaçoes dentro dessa coluna (sao varias informações sobre variantes separadas por ',')\n",
    " - hgvs_expressions = São uma espécie de códigos que aparecem no detalhe na variante, 'representative_transcript' faz parte dele também, nem todos possuem\n",
    " - last_review_date = me parece irrelevante assim como no df de genes\n",
    " - civic_variant_evidence_score = pontuação do perfil molecular, possui varios duplicados mas nenhum nulo\n",
    " - allele_registry_id = ID de registro de alelo (Os alelos são as formas alternativas, gerados por mutações, de um mesmo gene), nem todos possuem\n",
    " - clinvar_ids = ClinVar ID possui um código associado a um link (https://www.ncbi.nlm.nih.gov/clinvar/variation/12624/?new_evidence=true) com varias informações genéticas, varios nulos.\n",
    " - variant_aliases = mais uma vez vários nomes de variantes juntos, separados por virgula, varios nulos\n",
    " - os 2 abaixo são relacionados, quando acessado o link, aparece um diagnóstico.\n",
    " - assertion_ids = são valores únicos, apenas 31 valores são nulos, são codigos numericos, ao acessar o link o código faz conjunto com a sigla 'AID', exemploa abaixo 'AID20'\n",
    " - assertion_civic_urls = link de url para detalhes do diagnosticos, também possui 31 valores nulos\n",
    "https://civicdb.org/assertions/20/summary\n",
    " - is_flagged = facil detecção, não possui nulo, maioria FALSE, 1 TRUE, e o resto possui link do assertion(https://civicdb.org/assertions/10/summary)\n",
    " - As ultimas 4 colunas (Unnamed: 29\tUnnamed: 30\tUnnamed: 31\tUnnamed: 32), possuem apenas 10 linhas com valores, sendo elas 'FALSE' ou o link do assertion (https://civicdb.org/assertions/12/summary), maioria com valores nulos."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>drugs</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>evidence_level</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "      <th>source_type</th>\n",
       "      <th>asco_abstract_id</th>\n",
       "      <th>citation</th>\n",
       "      <th>nct_ids</th>\n",
       "      <th>rating</th>\n",
       "      <th>evidence_status</th>\n",
       "      <th>evidence_id</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "      <th>chromosome2</th>\n",
       "      <th>start2</th>\n",
       "      <th>stop2</th>\n",
       "      <th>representative_transcript2</th>\n",
       "      <th>ensembl_version</th>\n",
       "      <th>reference_build</th>\n",
       "      <th>variant_summary</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>evidence_civic_url</th>\n",
       "      <th>variant_civic_url</th>\n",
       "      <th>gene_civic_url</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>Negative</td>\n",
       "      <td>JAK2 V617F is not associated with lymphoid leu...</td>\n",
       "      <td>16081687</td>\n",
       "      <td>PubMed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Levine et al., 2005, Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>1</td>\n",
       "      <td>64</td>\n",
       "      <td>28</td>\n",
       "      <td>9</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000381652.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2020-11-20 16:18:21 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/64</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/28</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>5156</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>Negative</td>\n",
       "      <td>GIST tumors harboring PDGFRA D842V mutation ar...</td>\n",
       "      <td>15146165</td>\n",
       "      <td>PubMed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lasota et al., 2004, Lab. Invest.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>38</td>\n",
       "      <td>4</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000257290.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>PDGFRA D842 mutations are characterized broadl...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2015-06-21 16:49:38 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/99</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/38</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene  entrez_id variant                         disease    doid  \\\n",
       "0    JAK2       3717   V617F               Lymphoid Leukemia  1037.0   \n",
       "1  PDGFRA       5156   D842V  Gastrointestinal Stromal Tumor  9253.0   \n",
       "\n",
       "  phenotypes drugs drug_interaction_type evidence_type evidence_direction  \\\n",
       "0        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "1        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "\n",
       "  evidence_level clinical_significance  \\\n",
       "0              B              Negative   \n",
       "1              B              Negative   \n",
       "\n",
       "                                  evidence_statement  citation_id source_type  \\\n",
       "0  JAK2 V617F is not associated with lymphoid leu...     16081687      PubMed   \n",
       "1  GIST tumors harboring PDGFRA D842V mutation ar...     15146165      PubMed   \n",
       "\n",
       "   asco_abstract_id                           citation nct_ids  rating  \\\n",
       "0               NaN         Levine et al., 2005, Blood     NaN     4.0   \n",
       "1               NaN  Lasota et al., 2004, Lab. Invest.     NaN     3.0   \n",
       "\n",
       "  evidence_status  evidence_id  variant_id  gene_id chromosome       start  \\\n",
       "0        accepted            1          64       28          9   5073770.0   \n",
       "1        accepted            2          99       38          4  55152093.0   \n",
       "\n",
       "         stop reference_bases variant_bases representative_transcript  \\\n",
       "0   5073770.0               G             T         ENST00000381652.3   \n",
       "1  55152093.0               A             T         ENST00000257290.5   \n",
       "\n",
       "  chromosome2  start2  stop2 representative_transcript2  ensembl_version  \\\n",
       "0         NaN     NaN    NaN                        NaN             75.0   \n",
       "1         NaN     NaN    NaN                        NaN             75.0   \n",
       "\n",
       "  reference_build                                    variant_summary  \\\n",
       "0          GRCh37  JAK2 V617F is a highly recurrent mutation in m...   \n",
       "1          GRCh37  PDGFRA D842 mutations are characterized broadl...   \n",
       "\n",
       "  variant_origin         last_review_date  \\\n",
       "0        Somatic  2020-11-20 16:18:21 UTC   \n",
       "1        Somatic  2015-06-21 16:49:38 UTC   \n",
       "\n",
       "                                  evidence_civic_url  \\\n",
       "0  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "1  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "\n",
       "                                  variant_civic_url  \\\n",
       "0  https://civic.genome.wustl.edu/links/variants/64   \n",
       "1  https://civic.genome.wustl.edu/links/variants/99   \n",
       "\n",
       "                                  gene_civic_url  is_flagged  \n",
       "0  https://civic.genome.wustl.edu/links/genes/28       False  \n",
       "1  https://civic.genome.wustl.edu/links/genes/38       False  "
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_evidence.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### df_evidence\n",
    " - gene = nenhuma valor nulo, alguns se repetem mas não possui duplicado\n",
    " - entrez_id = id aparece na tabela gene e variant\n",
    " - variant = não possui valor nulo, é o mesmo valor que aparece no df_variants['variant']\n",
    " - disease = diagnostico, varios duplicados, apenas 23 nulos. (possivel alvo preditor)\n",
    " - doid = possui 32 nulos, id de disease.\n",
    " - phenotypes = também tem relação com diagnóstico, mas voltado a carascteristicas, grande maioria nulo.\n",
    " - drugs = medicação, grande parte não possui informação.\n",
    " - drug_interaction_type = tipo de medicação, grande parte possui informação e somente 3 opções.\n",
    " - evidence_type = (possivel alvo preditor), 6 variaveis e inclui possivel diagnostico, não possui valores nulos\n",
    " - evidence_direction = poucos nulos, se suportou ou não o medicamento.\n",
    " - evidence_level = niveis de A a E ( “A - Validated association”, “B - Clinical evidence”, “C - Case study”, “D - Preclinical evidence”, and “E - Inferential association”. ), nào possui nulos\n",
    " - clinical_significance - acredito ser a resposta clinica ao tratamento, poucos nulos\n",
    " - evidence_statement = explicando a variante para o resultado. nenhum nulo\n",
    " - citation_id = id da publicação\n",
    " - source_type = nenhum nulo, onde foi publicado\n",
    " - asco_abstract_id = quase todos nulos, registro da publicacao no asco.\n",
    " - citation = local da publicação\n",
    " - nct_ids = quase todos nulos, ID do ensaio clínico associado ao item de evidência\n",
    " - rating = poucos nulos, nota para o estudo\n",
    " - evidence_status = todos aceitos\n",
    " - evidence_id = numero de registro da evidencia, todos unicos\n",
    " - variant_id = id da variante\n",
    " - gene_id = id do gene\n",
    " - chromosome = em pesquisa me parece que cada cromossomo tem sua função, necessário entender melhor sobre\n",
    " \n",
    " abaixo, todos já respondidos anteriormente\n",
    " - start = \n",
    " - stop = \n",
    " - reference_bases = \n",
    " - variant_bases = \n",
    " - representative_transcript = \n",
    " - chromosome2 = \n",
    " - start2 = \n",
    " - stop2 = \n",
    " - representative_transcript2 = \n",
    " - ensembl_version =\n",
    " - reference_build =\n",
    " \n",
    " \n",
    " - variant_summary = muitos nulos, descrição da variante\n",
    " - variant_origin = como começou\n",
    " - last_review_date = irrelevante, será excluido\n",
    " - evidence_civic_url = link para a evidencia\n",
    " - variant_civic_url = link para a variante\n",
    " - gene_civic_url = link para o gene\n",
    " - is_flagged = facil detecção\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>drugs</th>\n",
       "      <th>assertion_type</th>\n",
       "      <th>assertion_direction</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>acmg_codes</th>\n",
       "      <th>amp_category</th>\n",
       "      <th>nccn_guideline</th>\n",
       "      <th>nccn_guideline_version</th>\n",
       "      <th>regulatory_approval</th>\n",
       "      <th>fda_companion_test</th>\n",
       "      <th>assertion_summary</th>\n",
       "      <th>assertion_description</th>\n",
       "      <th>assertion_id</th>\n",
       "      <th>evidence_item_ids</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>assertion_civic_url</th>\n",
       "      <th>evidence_items_civic_url</th>\n",
       "      <th>variant_civic_url</th>\n",
       "      <th>gene_civic_url</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERBB2</td>\n",
       "      <td>2064</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Her2-receptor Positive Breast Cancer</td>\n",
       "      <td>60079.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Trastuzumab</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tier I - Level A</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>3.2017</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>HER2 amplification predicts sensitivity to Tra...</td>\n",
       "      <td>HER2 amplification defines a clinically releva...</td>\n",
       "      <td>2</td>\n",
       "      <td>529,528,1122</td>\n",
       "      <td>18</td>\n",
       "      <td>20</td>\n",
       "      <td>2020-01-16 19:05:22 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/assertions/2</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/18</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/20</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ALK</td>\n",
       "      <td>238</td>\n",
       "      <td>Fusion</td>\n",
       "      <td>Lung Non-small Cell Carcinoma</td>\n",
       "      <td>3908.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crizotinib</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>Supports</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tier I - Level A</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>9.2017</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Lung adenocarcinoma positive for ALK-FUSIONS h...</td>\n",
       "      <td>ALK-FUSIONS induce an active form of the recep...</td>\n",
       "      <td>3</td>\n",
       "      <td>1419,1201,1200,4858,1199,1198,1356,1691,1187,1...</td>\n",
       "      <td>499</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-01-16 19:05:22 UTC</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/assertions/3</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/evidence_...</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variants/499</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/genes/1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene  entrez_id        variant                               disease  \\\n",
       "0  ERBB2       2064  Amplification  Her2-receptor Positive Breast Cancer   \n",
       "1    ALK        238         Fusion         Lung Non-small Cell Carcinoma   \n",
       "\n",
       "      doid phenotypes        drugs assertion_type assertion_direction  \\\n",
       "0  60079.0        NaN  Trastuzumab     Predictive            Supports   \n",
       "1   3908.0        NaN   Crizotinib     Predictive            Supports   \n",
       "\n",
       "  clinical_significance acmg_codes      amp_category  \\\n",
       "0  Sensitivity/Response        NaN  Tier I - Level A   \n",
       "1  Sensitivity/Response        NaN  Tier I - Level A   \n",
       "\n",
       "               nccn_guideline  nccn_guideline_version  regulatory_approval  \\\n",
       "0               Breast Cancer                  3.2017                 True   \n",
       "1  Non-Small Cell Lung Cancer                  9.2017                 True   \n",
       "\n",
       "   fda_companion_test                                  assertion_summary  \\\n",
       "0                True  HER2 amplification predicts sensitivity to Tra...   \n",
       "1                True  Lung adenocarcinoma positive for ALK-FUSIONS h...   \n",
       "\n",
       "                               assertion_description  assertion_id  \\\n",
       "0  HER2 amplification defines a clinically releva...             2   \n",
       "1  ALK-FUSIONS induce an active form of the recep...             3   \n",
       "\n",
       "                                   evidence_item_ids  variant_id  gene_id  \\\n",
       "0                                       529,528,1122          18       20   \n",
       "1  1419,1201,1200,4858,1199,1198,1356,1691,1187,1...         499        1   \n",
       "\n",
       "          last_review_date                                assertion_civic_url  \\\n",
       "0  2020-01-16 19:05:22 UTC  https://civic.genome.wustl.edu/links/assertions/2   \n",
       "1  2020-01-16 19:05:22 UTC  https://civic.genome.wustl.edu/links/assertions/3   \n",
       "\n",
       "                            evidence_items_civic_url  \\\n",
       "0  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "1  https://civic.genome.wustl.edu/links/evidence_...   \n",
       "\n",
       "                                   variant_civic_url  \\\n",
       "0   https://civic.genome.wustl.edu/links/variants/18   \n",
       "1  https://civic.genome.wustl.edu/links/variants/499   \n",
       "\n",
       "                                  gene_civic_url  is_flagged  \n",
       "0  https://civic.genome.wustl.edu/links/genes/20       False  \n",
       "1   https://civic.genome.wustl.edu/links/genes/1       False  "
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_assertions.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### df_assertions\n",
    "\n",
    " - gene = nome do gene, nenhum nulo mas alguns se repetem\n",
    " - entrez_id = aparece em todas as tabelas anteriores, nenhum nulo mas alguns se repetem\n",
    " - variant = nome da variante, nenhum nulo mas alguns se repetem\n",
    " - disease = diagnóstico, nenhum nulo mas alguns se repetem\n",
    " - doid = id de disease, 1 nulo\n",
    " - phenotypes = também tem relação com diagnóstico, mas voltado a carascteristicas, grande maioria nulo.\n",
    " - drugs = medicação\n",
    " - assertion_type = (possivel alvo preditor), 6 variaveis e inclui possivel diagnostico, não possui valores nulos\n",
    " - assertion_direction = se suportou ou não o medicamento.\n",
    " - clinical_significance = se suportou ou não o medicamento.\n",
    " - acmg_codes = código clínico\n",
    " - amp_category = apelido na medicação e nível\n",
    " - nccn_guideline = tipo do cancer\n",
    " - nccn_guideline_version = id do cancer\n",
    " - regulatory_approval = se o tratamento aprovado, todos preenchidos\n",
    " - fda_companion_test = aprovação do fds (tipo anvisa)\n",
    " - assertion_summary = breve descrição do tratamento e diagnostico\n",
    " - assertion_description = descrição do tratamento e diagnostico\n",
    " - assertion_id = id do assertion\n",
    " - evidence_item_ids = ids de evidence, mais de 1 (provavelmente este é o teste do treinamento)\n",
    " - variant_id = id da variante\n",
    " - gene_id = id do gene\n",
    " - last_review_date = irrelevante\n",
    " - assertion_civic_url = link para o assertion\n",
    " - evidence_items_civic_url = link para evidence\n",
    " - variant_civic_url = link para variant\n",
    " - gene_civic_url = link para gene\n",
    " - is_flagged = se deu sinais positivos ou nao"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_group_id</th>\n",
       "      <th>variant_group_civic_url</th>\n",
       "      <th>variant_group</th>\n",
       "      <th>description</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variant_g...</td>\n",
       "      <td>Imatinib Resistance</td>\n",
       "      <td>While imatinib has shown to be incredibly succ...</td>\n",
       "      <td>2015-06-21 16:49:55 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variant_g...</td>\n",
       "      <td>KIT Exon 17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2016-04-15 17:15:01 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>https://civic.genome.wustl.edu/links/variant_g...</td>\n",
       "      <td>Crizotinib Resistance</td>\n",
       "      <td>The ALK oncogene has long been considered a dr...</td>\n",
       "      <td>2015-06-21 16:49:55 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   variant_group_id                            variant_group_civic_url  \\\n",
       "0                 1  https://civic.genome.wustl.edu/links/variant_g...   \n",
       "1                 2  https://civic.genome.wustl.edu/links/variant_g...   \n",
       "2                 3  https://civic.genome.wustl.edu/links/variant_g...   \n",
       "\n",
       "           variant_group                                        description  \\\n",
       "0    Imatinib Resistance  While imatinib has shown to be incredibly succ...   \n",
       "1            KIT Exon 17                                                NaN   \n",
       "2  Crizotinib Resistance  The ALK oncogene has long been considered a dr...   \n",
       "\n",
       "          last_review_date  is_flagged  \n",
       "0  2015-06-21 16:49:55 UTC       False  \n",
       "1  2016-04-15 17:15:01 UTC       False  \n",
       "2  2015-06-21 16:49:55 UTC       False  "
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variantGroup.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### df_variantGroup\n",
    " - variant_group_id = index\n",
    " - variant_group_civic_url = possui link associado ao tratamento, nenhum valor nulo\n",
    " - variant_group = todos valores unicos e nenhum nulo, nome do tratamento\n",
    " - description = descrição do tratamento, alguns nulos\n",
    " - last_review_date = me parece irrelevante assim como no df de genes\n",
    " - is_flagged = não possui nulo e todos os valores são 'FALSE'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALT_2</th>\n",
       "      <th>ALT_3</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER_PASS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>8073509</td>\n",
       "      <td>691</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11177096</td>\n",
       "      <td>609</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11184573</td>\n",
       "      <td>542</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11184580</td>\n",
       "      <td>284</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>11187094</td>\n",
       "      <td>470</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS   ID REF ALT_1  ALT_2  ALT_3  QUAL  FILTER_PASS\n",
       "0     1   8073509  691   C     A    NaN    NaN   NaN        False\n",
       "1     1  11177096  609   C     A    NaN    NaN   NaN        False\n",
       "2     1  11184573  542   G     T    NaN    NaN   NaN        False\n",
       "3     1  11184580  284   G     A    NaN    NaN   NaN        False\n",
       "4     1  11187094  470   G     T    NaN    NaN   NaN        False"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "accepted_variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALT_2</th>\n",
       "      <th>ALT_3</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER_PASS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>8073509</td>\n",
       "      <td>691</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11177096</td>\n",
       "      <td>609</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11184573</td>\n",
       "      <td>542</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11184580</td>\n",
       "      <td>284</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>11187094</td>\n",
       "      <td>470</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS   ID REF ALT_1  ALT_2  ALT_3  QUAL  FILTER_PASS\n",
       "0     1   8073509  691   C     A    NaN    NaN   NaN        False\n",
       "1     1  11177096  609   C     A    NaN    NaN   NaN        False\n",
       "2     1  11184573  542   G     T    NaN    NaN   NaN        False\n",
       "3     1  11184580  284   G     A    NaN    NaN   NaN        False\n",
       "4     1  11187094  470   G     T    NaN    NaN   NaN        False"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "accepted_sub.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**acredito ser irrelevante**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [],
   "source": [
    "#juntando dados do df variant em evidence\n",
    "df_merge = df_evidence.merge(df_variants, left_on = 'variant_id', right_on = 'variant_id',\n",
    "                 how='left', suffixes=('', '_y'))\n",
    "\n",
    "df_merge.drop(df_merge.filter(regex='_y$').columns, axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "''''\n",
    "apenas se quiser adicionar a descrição dos genes, não possui para todos, para isso, necessário alterar nome da \n",
    "coluna da descrição para entendimento, assim como coluna name já possui no fd_evidence como coluna gene.\n",
    "\n",
    "por se tratar de dados sem relevancia optei por nao ter esse trabalho\n",
    "'''\n",
    "#df_merge = df_merge.merge(df_genes, left_on = 'gene_id', right_on = 'gene_id',\n",
    "#                 how='left', suffixes=('', '_y'))\n",
    "\n",
    "#df_merge.drop(df_merge.filter(regex='_y$').columns, axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3897\n",
      "3897\n"
     ]
    }
   ],
   "source": [
    "print(df_evidence.shape[0])\n",
    "print(df_merge.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene', 'entrez_id', 'variant', 'disease', 'doid', 'phenotypes',\n",
       "       'drugs', 'drug_interaction_type', 'evidence_type', 'evidence_direction',\n",
       "       'evidence_level', 'clinical_significance', 'evidence_statement',\n",
       "       'citation_id', 'source_type', 'asco_abstract_id', 'citation', 'nct_ids',\n",
       "       'rating', 'evidence_status', 'evidence_id', 'variant_id', 'gene_id',\n",
       "       'chromosome', 'start', 'stop', 'reference_bases', 'variant_bases',\n",
       "       'representative_transcript', 'chromosome2', 'start2', 'stop2',\n",
       "       'representative_transcript2', 'ensembl_version', 'reference_build',\n",
       "       'variant_summary', 'variant_origin', 'last_review_date',\n",
       "       'evidence_civic_url', 'variant_civic_url', 'gene_civic_url',\n",
       "       'is_flagged', 'summary', 'variant_groups', 'variant_types',\n",
       "       'hgvs_expressions', 'civic_variant_evidence_score',\n",
       "       'allele_registry_id', 'clinvar_ids', 'variant_aliases', 'assertion_ids',\n",
       "       'assertion_civic_urls', 'Unnamed: 29', 'Unnamed: 30', 'Unnamed: 31',\n",
       "       'Unnamed: 32', 'name', 'description'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
